### PR TITLE
Add test cases for infra servers

### PR DESCRIPTION
**Pull request by vsingh-msys**
### :nut_and_bolt: Description: What code changed, and why?

Infra proxy service's servers CRUD operations have been implemented but the test cases were remaining. Added test cases `TestServers` for infra servers functionalities.

### :chains: Related Resources
NA

### :+1: Definition of Done
- Cover the test cases for `CreateServer`, `GetServers`, `GetServer`, `DeleteServer` and `UpdateServer`.
- Minor code cleanup and good to have changes

### :athletic_shoe: How to Build and Test the Change
 - Goto dir `cd components/infra-proxy-service`
 - `make setup_docker_pg`
 - `make test`

This PR added the test cases for existing `servers` functionalities no need for the build.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/infra-proxy-service/server/orgs_test.go [hunk 1]
```diff
@@ -1074,10 +1074,3 @@ func cleanupOrg(ctx context.Context, t *testing.T, cl infra_proxy.InfraProxyClie
 	_, err := cl.DeleteOrg(context.Background(), &deleteReq)
 	require.NoError(t, err)
 }
-
-func cleanupServer(ctx context.Context, t *testing.T, cl infra_proxy.InfraProxyClient, serverID string) {
```
Comment (vsingh-msys) [inline components/infra-proxy-service/server/orgs_test.go]: Move `cleanupServer` method to [servers_test.go](https://github.com/chef/automate/pull/3248/files#diff-26345778d259836272877d139e0983a4R514)

File: components/infra-proxy-service/server/servers.go [hunk 2]
```diff
@@ -35,6 +35,11 @@ func (s *Server) CreateServer(ctx context.Context, req *request.CreateServer) (*
 		return nil, status.Error(codes.InvalidArgument, "must supply server fqdn")
 	}
 
+	if req.IpAddress == "" {
```
Comment (vsingh-msys) [inline components/infra-proxy-service/server/servers.go]: Added missing validation for `IpAddress`

File: components/infra-proxy-service/server/servers.go [hunk 3]
```diff
@@ -67,7 +72,7 @@ func (s *Server) GetServer(ctx context.Context, req *request.GetServer) (*respon
 
 	UUID, err := uuid.FromString(req.Id)
 	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "invalid server id: %s", err.Error())
+		return nil, status.Errorf(codes.InvalidArgument, "invalid server ID: %s", err.Error())
```
Comment (vsingh-msys) [inline components/infra-proxy-service/server/servers.go]: Minor good to have changes

File: components/infra-proxy-service/server/servers_test.go [hunk 4]
```diff
@@ -0,0 +1,519 @@
...
+				Description: "Chef infra server",
+				Fqdn:        "domain.com",
+				IpAddress:   "0.0.0.0",
+			})
+			assert.Nil(t, resp)
+			assert.Error(t, err, "must supply server name")
+			grpctest.AssertCode(t, codes.InvalidArgument, err)
+		})
+
+		t.Run("when the server name is already exist, raise invalid argument error", func(t *testing.T) {
```
Comment (msorens) [inline components/infra-proxy-service/server/servers_test.go:83]:
```suggestion
		t.Run("when the server name already exists, raise invalid argument error", func(t *testing.T) {
```

File: components/infra-proxy-service/server/servers_test.go [hunk 5]
```diff
@@ -0,0 +1,519 @@
...
+			require.NotNil(t, list)
+			assert.Contains(t, list.Servers, resp1.Server)
+			assert.Contains(t, list.Servers, resp2.Server)
+			assert.Equal(t, 2, len(list.Servers))
+
+			cleanupServer(ctx, t, cl, resp1.Server.Id)
+			cleanupServer(ctx, t, cl, resp2.Server.Id)
+		})
+
+		t.Run("when the server is exist with it's orgs, return servers list with orgs's count successfully", func(t *testing.T) {
```
Comment (msorens) [inline components/infra-proxy-service/server/servers_test.go:181]: Slight cleanup; also the word "successfully" is already implied so not needed.
```suggestion
		t.Run("when the server exists with orgs, return servers list with org count", func(t *testing.T) {
```

File: components/infra-proxy-service/server/servers_test.go [hunk 6]
```diff
@@ -0,0 +1,519 @@
...
+			assert.Equal(t, resp1.Server.Name, server1.Server.Name)
+			assert.Equal(t, resp1.Server.Description, server1.Server.Description)
+			assert.Equal(t, resp1.Server.Fqdn, server1.Server.Fqdn)
+			assert.Equal(t, resp1.Server.IpAddress, server1.Server.IpAddress)
+			assert.EqualValues(t, 0, server1.Server.OrgsCount)
+
+			cleanupServer(ctx, t, cl, resp1.Server.Id)
+		})
+
+		t.Run("when the server exists with it's orgs, return the server with orgs's count successfully", func(t *testing.T) {
```
Comment (msorens) [inline components/infra-proxy-service/server/servers_test.go:270]:
```suggestion
		t.Run("when the server exists with orgs, return the server with org count", func(t *testing.T) {
```

File: components/infra-proxy-service/server/servers_test.go [hunk 7]
```diff
@@ -0,0 +1,519 @@
...
+		t.Run("when the server ID for the server to delete is not valid, raise an invalid argument error", func(t *testing.T) {
+			resp, err := cl.GetServer(ctx, &request.GetServer{
+				Id: "35bffbab-3a49-dd8a-94a1-9ea87ec5c3cc",
+			})
+
+			require.Nil(t, resp)
+			grpctest.AssertCode(t, codes.InvalidArgument, err)
+		})
+
+		t.Run("when the server to delete is does not exist, returns server not found", func(t *testing.T) {
```
Comment (msorens) [inline components/infra-proxy-service/server/servers_test.go:356]:
```suggestion
		t.Run("when the server to delete does not exist, returns server not found", func(t *testing.T) {
```

File: components/infra-proxy-service/server/servers_test.go [hunk 8]
```diff
@@ -0,0 +1,519 @@
...
+				Name:        "chef-infra-server",
+				Description: "Chef infra server",
+				Fqdn:        "domain.com",
+			})
+			assert.Nil(t, resp)
+			assert.Error(t, err, "must supply server IP address")
+			grpctest.AssertCode(t, codes.InvalidArgument, err)
+		})
+
+		t.Run("when the server name for the server to update is already exists, raise invalid argument error", func(t *testing.T) {
```
Comment (msorens) [inline components/infra-proxy-service/server/servers_test.go:466]:
```suggestion
		t.Run("when the server name for the server to update already exists, raise invalid argument error", func(t *testing.T) {
```

**Pull request merged**